### PR TITLE
Task 65 header rework

### PR DESCRIPTION
**Pull request by sergevei**

### CODE REVIEW

File: src/containers/header/index.js [hunk 1]
```diff
@@ -4,26 +4,43 @@ import { NavLink } from 'react-router-dom';
...
+    }
+  });
+
+  const toggleMenuFunc = () => {
+    if (width < 768) {
+      const toggleMenu = document.querySelector('#menu');
+      toggleMenu.classList.toggle('mobileMenuActive');
+    }
+  };
+
```
Comment (melonholy) [inline src/containers/header/index.js:43]: Remove ctrl+c ctrl+v code

File: src/containers/header/index.js [hunk 2]
```diff
@@ -39,6 +56,15 @@ const Header = props => {
     setModalStatusReg(true);
   };
 
+  const checkMobileMenuStatus = () => {
+    toggleMenuFunc();
+    if (isOpenMenu) {
```
Comment (melonholy) [inline src/containers/header/index.js]: `setMenuStatus(!isOpenMenu)`

File: src/containers/header/styles.module.scss [hunk 3]
```diff
@@ -85,11 +95,92 @@
...
-      flex-basis: 20%;
-      padding: 10px 0px;
-      @media (max-width: $width-lg) {
-        flex-basis: 23%;
+      padding: 10px 20px;
+      transition: none;
+    }
+  }
+
+  @media (max-width: $width-md) {
```
Comment (melonholy) [inline src/containers/header/styles.module.scss]: Put media-queries in class block

File: src/containers/header/styles.module.scss [hunk 4]
```diff
@@ -85,11 +95,92 @@
...
+
+      div {
+        margin: 15px 0;
+        text-align: center;
       }
+
+      button {
+        width: 100%;
+        max-width: 200px;
+        margin-top: 5px;
```
Comment (melonholy) [inline src/containers/header/styles.module.scss]: Combine in one margin

**Review by melonholy — CHANGES_REQUESTED** [review]

File: src/containers/header/index.js (hunk 1 above)
Comment (OlegPanasyuk) [reply src/containers/header/index.js:43]: I think he needs just to push `const toggleMenu = document.querySelector('#menu');`
from the functions.

File: src/containers/header/menu.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.mobileMenuActive {
+  display: flow-root !important;
```
Comment (OlegPanasyuk) [inline src/containers/header/menu.scss:2]: Doesn't it work without `!important`?
I knew that the `!important` is a bad practice.

File: src/containers/header/index.js [hunk 6]
```diff
@@ -48,68 +74,73 @@ const Header = props => {
...
+        onModalClose={setModalForgotPass}
+      />
     </header>
   );
 };
 
 Header.propTypes = {
   logOutRequest: PropTypes.func.isRequired,
-  isAuthenticated: PropTypes.bool
+  isAuthenticated: PropTypes.bool,
```
Comment (OlegPanasyuk) [inline src/containers/header/index.js]: Make it required or add to code default values block

File: src/containers/profile/profileTab/index.jsx [hunk 7]
```diff
@@ -179,9 +179,11 @@ const ProfileTab = ({ profile, getProfileRequest, updateProfileRequest }) => {
         </div>
         <div className={styles.buttonsBlock}>
           {checked ? (
-            <button type="button" onClick={onCheck}>
-              Update
-            </button>
+            <div>
```
Comment (OlegPanasyuk) [inline src/containers/profile/profileTab/index.jsx]: What did you use this block for?

**Review by OlegPanasyuk — APPROVED** [review]

File: src/containers/header/index.js [hunk 8]
```diff
@@ -4,26 +4,43 @@ import { NavLink } from 'react-router-dom';
 
 import PropTypes from 'prop-types';
 import { MdSearch } from 'react-icons/md';
+import { FaBars, FaWindowClose } from 'react-icons/fa';
```
Comment (OlegPanasyuk) [inline src/containers/header/index.js:7]: !

**Review by OlegPanasyuk — CHANGES_REQUESTED** [review]

File: src/containers/header/index.js [hunk 9]
```diff
@@ -4,26 +4,43 @@ import { NavLink } from 'react-router-dom';
...
+  const [isOpenMenu, setMenuStatus] = useState(false);
   const [isOpenForgotPass, setModalForgotPass] = useState(false);
 
   useEffect(() => {
     const handleWidth = () => setWidth(window.innerWidth);
     window.addEventListener('resize', handleWidth);
   }, []);
 
+  useEffect(() => {
+    if (width > 768) {
```
Comment (universezzz) [inline src/containers/header/index.js:30]: width >= 768

File: src/containers/header/index.js [hunk 10]
```diff
@@ -4,26 +4,43 @@ import { NavLink } from 'react-router-dom';
...
 
 const Header = props => {
   const [width, setWidth] = useState(window.innerWidth);
   const [isOpenLog, setModalStatusLog] = useState(false);
   const [isOpenReg, setModalStatusReg] = useState(false);
+  const [isOpenMenu, setMenuStatus] = useState(false);
   const [isOpenForgotPass, setModalForgotPass] = useState(false);
 
   useEffect(() => {
     const handleWidth = () => setWidth(window.innerWidth);
```
Comment (universezzz) [inline src/containers/header/index.js:25]: image
image
Your handler will work on every pixel. 
resize 1440px -> 440px  = 1k function calls

File: src/containers/header/index.js [hunk 11]
```diff
@@ -4,26 +4,43 @@ import { NavLink } from 'react-router-dom';
...
     window.addEventListener('resize', handleWidth);
   }, []);
 
+  useEffect(() => {
+    if (width > 768) {
+      const toggleMenu = document.querySelector('#menu');
```
Comment (universezzz) [inline src/containers/header/index.js]: You can move out this variable from two functions.
image

File: src/containers/header/menu.scss (hunk 5 above)
Comment (sergevei) [reply src/containers/header/menu.scss:2]: It doesn't work without !important

File: src/containers/header/index.js (hunk 6 above)
Comment (sergevei) [reply src/containers/header/index.js]: done

**Review by slavasgod — APPROVED** [review]

**Review by universezzz — APPROVED** [review]

**Pull request merged**